core/include/kernel/interrupt.h

```diff
@@ -32,17 +32,24 @@ enum itr_return {
 	ITRR_HANDLED,
 };
 
+struct itr_handler;
+
+typedef enum itr_return (*itr_handler_t)(struct itr_handler *h);
```

Comment:
We avoid typedefs as much as possible (coding rule, enforced by checkpatch). This one doesn't seem particularly useful since it is used only once locally, right?